<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>318</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="start">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>10</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>start</string>
    </property>
   </widget>
   <widget class="QPushButton" name="stop">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>40</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>stop</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="in_data1">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>10</y>
      <width>69</width>
      <height>26</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-999999999999999983222784.000000000000000</double>
    </property>
    <property name="maximum">
     <double>1000000000000000019884624838656.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>in_data1</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="in_data2">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>40</y>
      <width>69</width>
      <height>26</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-999999999999999983222784.000000000000000</double>
    </property>
    <property name="maximum">
     <double>1000000000000000019884624838656.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>40</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>in_data2</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="in_data3">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>70</y>
      <width>69</width>
      <height>26</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-999999999999999983222784.000000000000000</double>
    </property>
    <property name="maximum">
     <double>1000000000000000019884624838656.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>70</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>in_data3</string>
    </property>
   </widget>
   <widget class="QLed" name="thread_status">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>210</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Binary Led</string>
    </property>
    <property name="whatsThis">
     <string>Led widget</string>
    </property>
    <property name="value">
     <bool>false</bool>
    </property>
    <property name="onColor">
     <enum>QLed::Green</enum>
    </property>
    <property name="offColor">
     <enum>QLed::Grey</enum>
    </property>
    <property name="shape">
     <enum>QLed::Circle</enum>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>100</y>
      <width>271</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>150</y>
      <width>71</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>out_data2</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="out_data1">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>120</y>
      <width>69</width>
      <height>26</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::NoButtons</enum>
    </property>
    <property name="minimum">
     <double>-999999999999999983222784.000000000000000</double>
    </property>
    <property name="maximum">
     <double>1000000000000000019884624838656.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>120</y>
      <width>71</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>out_data1</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="out_data2">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>150</y>
      <width>69</width>
      <height>26</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::NoButtons</enum>
    </property>
    <property name="minimum">
     <double>-999999999999999983222784.000000000000000</double>
    </property>
    <property name="maximum">
     <double>1000000000000000019884624838656.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>210</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>thread in progress</string>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>213</x>
      <y>110</y>
      <width>20</width>
      <height>131</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>120</y>
      <width>161</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>video devices available</string>
    </property>
   </widget>
   <widget class="QListWidget" name="video_devices_list">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>140</y>
      <width>161</width>
      <height>101</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>400</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QLed</class>
   <extends>QWidget</extends>
   <header>qled.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
